s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/utils/AWSV2SourceClient.java

```diff
@@ -85,6 +85,10 @@ public Iterator<String> getListOfObjectKeys(final String startToken) {
         if (StringUtils.isNotBlank(startToken)) {
             request.withStartAfter(startToken);
         }
+        // Prefix is optional so only use if supplied
+        if (StringUtils.isNotBlank(s3SourceConfig.getAwsS3Prefix())) {
+            request.withPrefix(s3SourceConfig.getAwsS3Prefix());
```

Comment:
do we have to include this check inside this if block ? if (response.isTruncated()) { ?
Can we also improvise that if block abit in similar way ?

Reply:
or can we reuse that request obj

Reply:
No the Continuation token does all the work we don't need to add additional information to the request obj.

Reply:
Prefix check is applied on here .iterate(s3Client.listObjectsV2(request) and not when response is truncated. 
however max keys and all other checks are applied in that block except this.

Reply:
So I checked the API and it doesn't look like we need to supply this information.
maxKeys tells the API how many additional records to retrieve from the initial request which is stored in AWS. 
The continuation token tells the S3 api which records to return based on the initial request and should not need any additional information to be included.

I'm actually also pretty sure that withBucketName in that request can be removed as well.
AWS example under Manual Pagination here.
https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/pagination.html

Reply:
in that case can we try removing max keys and bucket name too. probably it should work

Reply:
done